python/cudf/cudf/core/column/column.py

```diff
@@ -1360,6 +1339,31 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             elif np.issubdtype(data.dtype, np.datetime64):
                 data = data.fillna(np.datetime64("NaT"))
 
+    elif hasattr(arbitrary, "__array_interface__"):
+        # CUDF assumes values are always contiguous
+        desc = arbitrary.__array_interface__
+        shape = desc["shape"]
+        arb_dtype = np.dtype(desc["typestr"])
+        # CUDF assumes values are always contiguous
+        if len(shape) > 1:
+            raise ValueError("Data must be 1-dimensional")
+
+        if desc["strides"] is not None:
+            arbitrary = np.ascontiguousarray(arbitrary)
+
+        if dtype is not None:
+            arbitrary = arbitrary.astype(dtype)
```

Comment:
`arbitrary` doesn't necessarily expose an `astype` method here since it's not guaranteed to be a numpy array

Reply:
May want to do something like `arbitrary = np.asarray(arbitrary)` since it can handle it zero copy.